### PR TITLE
Notify user the selected Android device is not GPU profiling compatible.

### PR DESCRIPTION
**Pull request by lpy**
If a device is not GPU profiling compatible, pop up a dialog to notify users.

BUG: #3086

### CODE REVIEW

**Comment by lpy** [comment]
This is checked by Jay already, I need to clean up the code a little bit

**Comment by lpy** [comment]
alright, ptal :)

**Comment by lpy** [comment]
Here's the screenshot image

**Comment by lpy** [comment]
image

File: gapic/src/main/com/google/gapid/views/TracerDialog.java [hunk 1]
```diff
@@ -289,6 +295,7 @@ protected void buttonPressed(int buttonId) {
       private final Label fileLabel;
       private final Label pcsWarning;
       private final Label requiredFieldMessage;
+      private final Label gpuProfilingCapWarning;
```
[the remark this answers is not in the document]
Comment (hevrard) [reply gapic/src/main/com/google/gapid/views/TracerDialog.java]: I see "Cap" is already used in other names, like getPerfettoCaps. Maybe use "Caps" in all names as a diminutive of "Capability"?
I'll leave this decision up to you!

**Comment by lpy** [comment]
Here's how it looks like:
image

File: gapic/src/main/com/google/gapid/views/TracerDialog.java [hunk 2]
```diff
@@ -63,13 +63,16 @@
 import com.google.gapid.widgets.Widgets;
 
 import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.jface.dialogs.MessageDialog;
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/views/TracerDialog.java]: This import is no longer needed

File: gapic/src/main/com/google/gapid/views/TracerDialog.java [hunk 3]
```diff
@@ -256,10 +259,13 @@ protected void buttonPressed(int buttonId) {
           "NOTE: Mid-Execution capture for %s is experimental";
       private static final int DEFAULT_START_FRAME = 100;
       private static final String PERFETTO_LABEL = "Profile Config: ";
+      private static final String NO_GPU_PROFILING_CAPABILITY = "Warning: Selected device has no GPU profiling capability.";
 
       private final String date = TRACE_DATE_FORMAT.format(new Date());
 
       private List<DeviceCaptureInfo> devices;
+      private DeviceCaptureInfo selectedDevice;
+      private TraceType selectedTraceType;
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/views/TracerDialog.java]: Neither var is referenced anywhere

File: gapic/src/main/com/google/gapid/views/TracerDialog.java [hunk 4]
```diff
@@ -63,13 +63,16 @@
...
+import org.eclipse.jface.dialogs.MessageDialog;
 import org.eclipse.jface.viewers.ArrayContentProvider;
 import org.eclipse.jface.viewers.ComboViewer;
 import org.eclipse.jface.viewers.IStructuredSelection;
 import org.eclipse.jface.viewers.LabelProvider;
 import org.eclipse.jface.viewers.StructuredSelection;
 import org.eclipse.jface.window.Window;
 import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.KeyAdapter;
+import org.eclipse.swt.events.KeyEvent;
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/views/TracerDialog.java]: unused imports

**Comment by lpy** [comment]
PTAL, moved the warning down as suggested. I will filed a follow-up bug

File: gapic/src/main/com/google/gapid/views/TracerDialog.java [hunk 5]
```diff
@@ -443,6 +445,12 @@ protected void configureDialog(DirectoryDialog dialog) {
         requiredFieldMessage.setForeground(getDisplay().getSystemColor(SWT.COLOR_RED));
         requiredFieldMessage.setVisible(false);
 
+        gpuProfilingCapabilityWarning = withLayoutData(
+            createLabel(this, ""),
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/views/TracerDialog.java]: `createLabel(this, NO_GPU_PROFILING_CAPABILITY)`

and remove the setText below. `setText` typically requires a relayout, you are just getting lucky that it wasn't needed.

**Pull request merged**